interview-questions.md

```diff
@@ -7,64 +7,82 @@ Then, research the question to expand on your answer. Even if you feel you have
+  Researched answer: A function is a block of organized, reusable code that is used to perform a single, related action. 
+  
+  (https://www.tutorialspoint.com/computer_programming/computer_programming_functions.htm#:~:text=Advertisements,high%20degree%20of%20code%20reusing.)
 
 
 
 2. What is the difference between .map() and .filter()?
 
-  Your answer:
+  Your answer: .map() allows users to reformat all index values of an array, whereas .filter() takes out certain index values.
 
-  Researched answer:
+  Researched answer: They do different things: . filter() returns a subset of the elements from the original array, while . map() produces an array with new, different entries based on the elements in the original array. 
+  
+  (https://discuss.codecademy.com/t/what-s-the-difference-between-filter-and-map/342066)
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: console.log() prints out the results of the called function on the terminal, whereas return gives you back a value that you are specifically looking for and is stored in the function. 
 
-  Researched answer:
+  Researched answer: Return is used to interrupt the flow of execution of a function and return some value (or not). console.log() is used to print information to the console (acessible by F12) of the browser. 
+  
+  (https://www.quora.com/What-is-the-difference-between-console-log-and-return-in-JavaScript#:~:text=Difference%20between%20console.-,log%20and%20return%20in%20javascript%3F,by%20F12)%20of%20the%20browser.)
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: An argument is what you run through the function to reformat.
 
-  Researched answer:
+  Researched answer: Arguments are the values or variables that you pass to a function when calling it.
+
+(https://www.quora.com/What-does-an-Argument-in-Programming-mean)
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Give each person an equal amount of work to do. Communicate communicate communicate. Don't be afraid of each other.
+               Navigator - the one who reads the instructions and tells the driver what to write down.
+               Driver - the one who listens to the navigator and writes the code down.
+
+  Researched answer: Pair programming is an agile software development technique in which two programmers work together at one workstation. One, the driver, writes code while the other, the observer or navigator, reviews each line of code as it is typed in. The two programmers switch roles frequently.
 
-  Researched answer:
+  (https://en.wikipedia.org/wiki/Pair_programming#:~:text=Pair%20programming%20is%20an%20agile,two%20programmers%20switch%20roles%20frequently.)
 
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
-  Your answer:
+  Your answer: I don't remember what TDD is or stands for...
```

Comment:
Thats ok! 

You'll hear this later on when dealing with interview questions, but if you dont know the answer, honesty is definitely the best policy, but there is a way to back it up and come out on top- "I'm not sure, but I know I can look it up at .. and .. " replace the blank spots where you would say perhaps your notes, or your syllabus, or official documentation or what you would google! 

This would show a proactiveness, even if you don't know the answer right off the bat, to finding the answer and that is super valuable to a almost every company.